crates/db_schema/src/impls/comment.rs

```diff
@@ -38,40 +32,71 @@ use url::Url;
 impl Comment {
   pub async fn permadelete_for_creator(
     pool: &mut DbPool<'_>,
-    for_creator_id: PersonId,
-  ) -> Result<Vec<Self>, Error> {
+    creator_id: PersonId,
+  ) -> LemmyResult<Vec<Self>> {
     let conn = &mut get_conn(pool).await?;
 
-    diesel::update(comment::table.filter(comment::creator_id.eq(for_creator_id)))
+    diesel::update(comment::table.filter(comment::creator_id.eq(creator_id)))
       .set((
         comment::content.eq(DELETED_REPLACEMENT_TEXT),
         comment::deleted.eq(true),
         comment::updated.eq(Utc::now()),
       ))
       .get_results::<Self>(conn)
       .await
+      .with_lemmy_type(LemmyErrorType::CouldntUpdateComment)
   }
 
   pub async fn update_removed_for_creator(
     pool: &mut DbPool<'_>,
-    for_creator_id: PersonId,
+    creator_id: PersonId,
     removed: bool,
-  ) -> Result<Vec<Self>, Error> {
+  ) -> LemmyResult<Vec<Self>> {
     let conn = &mut get_conn(pool).await?;
-    diesel::update(comment::table.filter(comment::creator_id.eq(for_creator_id)))
+    diesel::update(comment::table.filter(comment::creator_id.eq(creator_id)))
       .set((
         comment::removed.eq(removed),
         comment::updated.eq(Utc::now()),
       ))
       .get_results::<Self>(conn)
       .await
+      .with_lemmy_type(LemmyErrorType::CouldntUpdateComment)
+  }
+
+  pub async fn update_removed_for_creator_and_community(
+    pool: &mut DbPool<'_>,
+    creator_id: PersonId,
+    community_id: CommunityId,
+    removed: bool,
+  ) -> LemmyResult<Vec<CommentId>> {
+    let conn = &mut get_conn(pool).await?;
+
+    // Diesel can't update from join unfortunately, so you'll need to loop over these
+    let comment_ids = comment::table
+      .inner_join(post::table)
+      .filter(comment::creator_id.eq(creator_id))
+      .filter(post::community_id.eq(community_id))
+      .select(comment::id)
+      .load::<CommentId>(conn)
+      .await?;
+
+    let form = &CommentUpdateForm {
+      removed: Some(removed),
+      ..Default::default()
+    };
+
+    for comment_id in &comment_ids {
+      Comment::update(&mut conn.into(), *comment_id, form).await?;
+    }
```

Comment:
This query is extremely inefficient if there are more than a dozen comments. It would be much more efficient with `update(comment::table).filter(comment_id.eq_any(comment_ids)).set(form)`

Reply:
Done.